Description: Replace already obsolate gtk2 API's to new ones.
Forwarded: https://mail.gna.org/public/kanatest-friends/2012-03/msg00003.html
Bug-Debian: http://bugs.debian.org/634350
Author: Takahide Nojima <nozzy123nozzy@gmail.com>

--- a/src/gui.c
+++ b/src/gui.c
@@ -121,7 +121,8 @@
     GUI *appGUI = (GUI *)user_data;
 
     appGUI->tst->test_state = TRUE;
-    g_timeout_add (1000, time_handler, appGUI);
+
+    g_timeout_add (1000, (GSourceFunc)time_handler, appGUI);
 
     gtk_widget_show (appGUI->char_label);
     gtk_widget_hide (appGUI->logo_area);
@@ -626,7 +627,11 @@
     gtk_container_add (GTK_CONTAINER (frame), alignment);
     gtk_alignment_set_padding (GTK_ALIGNMENT (alignment), 0, 0, 12, 0);
 
+#if GTK_CHECK_VERSION(2,24,0)
+    appGUI->combobox_kana_mode = gtk_combo_box_text_new ();
+#else
     appGUI->combobox_kana_mode = gtk_combo_box_new_text ();
+#endif
     gtk_widget_show (appGUI->combobox_kana_mode);
     g_signal_connect (G_OBJECT (appGUI->combobox_kana_mode), "changed",
                       G_CALLBACK (gui_combobox_kana_handler_cb), NULL);
@@ -634,7 +639,11 @@
     gtk_container_set_border_width (GTK_CONTAINER (alignment), 4);
 
     for(i=0; i < KANA_MODE_NAMES; i++) {
+#if GTK_CHECK_VERSION(2,24,0)
+      gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT(appGUI->combobox_kana_mode), get_test_mode_name(i+1));
+#else
         gtk_combo_box_append_text (GTK_COMBO_BOX (appGUI->combobox_kana_mode), get_test_mode_name(i+1));
+#endif
     }
 
     appGUI->label_ka = gtk_label_new (NULL);
@@ -653,7 +662,11 @@
     gtk_container_add (GTK_CONTAINER (frame), alignment);
     gtk_alignment_set_padding (GTK_ALIGNMENT (alignment), 0, 0, 12, 0);
 
+#if GTK_CHECK_VERSION(2,24,0)
+    appGUI->combobox_lesson = gtk_combo_box_text_new ();
+#else
     appGUI->combobox_lesson = gtk_combo_box_new_text ();
+#endif
     gtk_widget_show (appGUI->combobox_lesson);
     g_signal_connect (G_OBJECT (appGUI->combobox_lesson), "changed",
                       G_CALLBACK (gui_combobox_kana_set_handler_cb), NULL);
@@ -661,7 +674,11 @@
     gtk_container_set_border_width (GTK_CONTAINER (alignment), 4);
 
     for(i=0; i < KANA_SET_NAMES; i++) {
+#if GTK_CHECK_VERSION(2,24,0)
+        gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (appGUI->combobox_lesson), get_kana_set_name(i));
+#else
         gtk_combo_box_append_text (GTK_COMBO_BOX (appGUI->combobox_lesson), get_kana_set_name(i));
+#endif
     }
 
     appGUI->label_le = gtk_label_new (NULL);
--- a/src/options.c
+++ b/src/options.c
@@ -841,8 +841,11 @@
     gtk_table_attach (GTK_TABLE (table), label, 0, 1, 0, 1,
                     (GtkAttachOptions) (GTK_FILL),
                     (GtkAttachOptions) (0), 0, 0);
-
+#if GTK_CHECK_VERSION(2,24,0)
+    combobox_ca_timeout = gtk_combo_box_text_new ();
+#else
     combobox_ca_timeout = gtk_combo_box_new_text ();
+#endif
 #if (GTK_MINOR_VERSION >= 22)
     gtk_widget_set_can_focus (combobox_ca_timeout, FALSE);
 #else
@@ -854,12 +857,19 @@
     gtk_table_attach (GTK_TABLE (table), combobox_ca_timeout, 1, 2, 0, 1,
                       (GtkAttachOptions) (GTK_FILL | GTK_EXPAND),
                       (GtkAttachOptions) (0), 8, 0);
+#if GTK_CHECK_VERSION(2,24,0)
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_ca_timeout), _("Disabled"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_ca_timeout), _("1 sec"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_ca_timeout), _("2 sec"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_ca_timeout), _("3 sec"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_ca_timeout), _("Until any key will be pressed"));
+#else
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_ca_timeout), _("Disabled"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_ca_timeout), _("1 sec"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_ca_timeout), _("2 sec"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_ca_timeout), _("3 sec"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_ca_timeout), _("Until any key will be pressed"));
-
+#endif
     gtk_combo_box_set_active (GTK_COMBO_BOX (combobox_ca_timeout), config.ca_timeout);
 
     /*-----------------------------------------------------------------*/
@@ -1082,8 +1092,11 @@
     gtk_box_pack_start (GTK_BOX (hbox), label, FALSE, FALSE, 0);
     gtk_label_set_use_markup (GTK_LABEL (label), TRUE);
     gtk_misc_set_padding (GTK_MISC (label), 4, 0);
-
+#if GTK_CHECK_VERSION(2,24,0)
+    combobox_dtf = gtk_combo_box_text_new ();
+#else
     combobox_dtf = gtk_combo_box_new_text ();
+#endif
 #if (GTK_MINOR_VERSION >= 22)
     gtk_widget_set_can_focus (combobox_dtf, FALSE);
 #else
@@ -1093,9 +1106,15 @@
     gtk_box_pack_start (GTK_BOX (hbox), combobox_dtf, FALSE, FALSE, 0);
     g_signal_connect (G_OBJECT (combobox_dtf), "changed",
                       G_CALLBACK (options_combobox_dtf_handler_cb), appGUI);
+#if GTK_CHECK_VERSION(2,24,0)
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_dtf), _("Romaji"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_dtf), _("Hiragana"));
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (combobox_dtf), _("Katakana"));
+#else
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_dtf), _("Romaji"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_dtf), _("Hiragana"));
     gtk_combo_box_append_text (GTK_COMBO_BOX (combobox_dtf), _("Katakana"));
+#endif
     gtk_combo_box_set_active (GTK_COMBO_BOX (combobox_dtf), 0);
 
     hbuttonbox_s = gtk_hbutton_box_new ();
