Description: Fix up gotcha according to Debian Policy. 
Author: Takahide Nojima <nozzy123nozzy@gmail.com>

--- a/Imakefile
+++ b/Imakefile
@@ -34,7 +34,8 @@
 EXTRACCOPTIONS = -Wall -Wpointer-arith -Wwrite-strings -Wtraditional
 #endif /* HasGcc */
 #if defined(LinuxArchitecture)
-#define STATICDEBUG -static
+/* #define STATICDEBUG -static */
+#define STATICDEBUG
 XCOMM XINTL=-lXintl
 #else
 #define STATICDEBUG
@@ -57,10 +58,11 @@
 XCOMM system environment
 XCOMM
 XCOMM don't you have gettimeofday?
-TIME =
+TIME = -DPOSIX
 XCOMM TIME = -DUSETIME
 XCOMM don't you have lockf or flock?
-LOCK =
+XCOMM LOCK =
+XCOMM for debianize
 LOCK = -DUSELOCKFILE
 
 XCOMM game defaults, you probably only want to alter the SCORE
@@ -69,9 +71,11 @@
 /* #define thefont "-*-courier-*-r-*-*-18-*-*-*-m-*-*-*" */
 XCOMM where is the high score file?
 XCOMM SCORE = $(LOCAL)/lib/$(APP)
-SCORE = /var/lib/games/$(APP)
-/* to install default gardens not in app_defaults/$(APP)
-/* #define InstallDefaultGardens $(SCORE)/gardens */
+XCOMM for debian policy manual
+SCORE = /var/games/$(APP)
+#define InstallDefaultScore $(DESTDIR)$(SCORE)
+/* to install default gardens not in app_defaults/$(APP) */
+#define InstallDefaultGardens InstallDefaultScore/gardens
 
 XCOMM The following defines are needed for some implementations and/or sites.
 XCOMM It is xmkmf's job to get these right in the first place.
@@ -91,11 +95,13 @@
 XCOMM
 XCOMM owner and access rights for executable & score directory
 XCOMM these are set for a setuid xmris
-OWNER = -o games
-INSTSCOREFLAGS = $(OWNER) -m 755
+OWNER = -o root -g games
+INSTUIDFLAGS=-m 2755
+INSTSCOREFLAGS = $(OWNER) -m 770
 INSTMRISFLAGS = $(OWNER) $(INSTUIDFLAGS)
 XCOMM normal install locations
-THEBINDIR = $(BINDIR)
+XCOMM for debian-policy manual.
+THEBINDIR = /usr/games
 THEMANDIR = $(MANDIR)
 XCOMM for open access you want 777 and $(INSTPGMFLAGS),
 XCOMM (the personal score files go in users' home dirs.)
@@ -112,12 +118,12 @@
 XLSFONT = xlsfont
 #endif /* thefont */
 FONT = thefont
-
-#ifndef InstallDefaultGardens
-#ifdef InstallAppDefFiles
-#define InstallDefaultGardens $(DESTDIR)/$(XAPPLOADDIR)/$(APP)
-#endif
-#endif
+XCOMM for debianize
+XCOMM #ifndef InstallDefaultGardens
+XCOMM #ifdef InstallAppDefFiles
+XCOMM #define InstallDefaultGardens $(DESTDIR)/$(XAPPLOADDIR)/$(APP)
+XCOMM #endif
+XCOMM #endif
 
 DEFINES = $(EXTRACCOPTIONS) -DFONT_NAME='$(FONT)' $(NDEBUG) \
 	-DSCORE_DIR='"$(SCORE)"' $(LOCK) $(TIME) $(KNR) -DSCROLLBARBUGFIX
@@ -238,9 +244,10 @@
 
 InstallProgramWithFlags(xmris,$(THEBINDIR),$(INSTMRISFLAGS))
 
-install :: $(THEBINDIR)/xmris
-	$(RM) $(THEBINDIR)/xmsit
-	$(LN) ./xmris $(THEBINDIR)/xmsit
+XCOMM commented out for debianize.
+XCOMM install :: $(THEBINDIR)/xmris
+XCOMM	$(RM) $(THEBINDIR)/xmsit
+XCOMM	$(LN) ./xmris $(THEBINDIR)/xmsit
 
 XCOMM xmred build
 NormalProgramTarget(xmred,$(MREDOBJS) $(WIDGETOBJS),$(DEPLIBS),$(LOCAL_LIBRARIES) -lm,)
@@ -270,16 +277,16 @@
 
 XCOMM score directory install
 install ::
-	MakeDir($(SCORE))
+	MakeDir(InstallDefaultScore)
 	@#not all $(INSTALL) have a -d flag, have to do it explicitly
-	@echo installing $(INSTSCOREFLAGS) $(SCORE)
+	@echo installing $(INSTSCOREFLAGS) InstallDefaultScore
 	@set -- $(INSTSCOREFLAGS) ; \
 	while [ $$# -ne 0 ] ; \
 	do \
 	case "$$1" in \
-	"-o") shift ; chown $$1 $(SCORE) ; echo owner is $$1 ;; \
-	"-m") shift ; chmod $$1 $(SCORE) ;; \
-	"-g") shift ; chgrp $$1 $(SCORE) ; echo group is $$1 ;; \
+	"-o") shift ; chown $$1 InstallDefaultScore ; echo owner is $$1 ;; \
+	"-m") shift ; chmod $$1 InstallDefaultScore ;; \
+	"-g") shift ; chgrp $$1 InstallDefaultScore ; echo group is $$1 ;; \
 	*) echo unknown option '$$1' ;; \
 	esac ; \
 	shift ; \
--- a/scoring.c
+++ b/scoring.c
@@ -31,9 +31,7 @@
 #else
 #include <pwd.h>
 #include <sys/stat.h>
-#ifdef USELOCKFILE
 #include <fcntl.h>
-#endif /* USELOCKFILE */
 #endif /* TRANSPUTER */
 /*}}}*/
 /*{{{  file locking*/
@@ -168,7 +166,6 @@
  */
 {
   FILE    *stream;
-
 #ifdef TRANSPUTER
   if(locking_file && !locks)
     /*{{{  attempt file lock*/
@@ -208,7 +205,7 @@
 	{
 	  if(current_uid != effective_uid)
 	    set_euid((current_uid = effective_uid));
-	  filed = open(locking_file, O_CREAT | O_EXCL, 0666);
+	  filed = open(locking_file, O_CREAT | O_EXCL, 0660);
 	  if(filed >= 0)
 	    break;
 	  if(errno == EINTR)
@@ -259,7 +256,7 @@
 #endif /* USELOCKFILE */
 #ifndef TRANSPUTER
   if(stream && flag & 4 && effective_uid == real_uid)
-    chmod(name, 0666); /* not everyone has fchmod */
+    chmod(name, 0660); /* not everyone has fchmod */
   if(current_uid != real_uid)
     set_euid((current_uid = real_uid));
 #endif /* TRANSPUTER */
--- a/xmred.man
+++ b/xmred.man
@@ -1,13 +1,13 @@
-. Copyright (C) 1993 Nathan Sidwell
-. RCS $Id: xmred.man 1.3 Tue, 16 Mar 1999 11:28:16 +0000 nathan $
+.\" . Copyright (C) 1993 Nathan Sidwell
+.\" RCS $Id: xmred.man 1.3 Tue, 16 Mar 1999 11:28:16 +0000 nathan $
 .TH XMRED 6 "20 December 1993" "X Version 11"
-.IX xmred#(n) "" "\fLxmred\fP(n)"
+.\" .IX xmred#(n) "" "\fLxmred\fP(n)"
 .SH NAME
 xmred - garden editor for xmris
 .SH SYNOPSIS
 .B xmred
 [-option ...] [-toolkitoption ...]
-.IX xmred#(n) "" "\fLxmred\fP(n) \(em video game editor"
+.\" .IX xmred#(n) "" "\fLxmred\fP(n) \(em video game editor"
 .SH DESCRIPTION
 .PP
 Mr Ed is a tool for creating and editing alternate gardens for Mr Is.
@@ -70,7 +70,7 @@
 of a centre and four edges. Depending on the exact location of the pointer,
 you will select either the centre, or an edge. Filling a path will place a
 random apple in the filled location. The four locations immediately below the
-'EXTRA' letters must be clear.
+\fIEXTRA\fR letters must be clear.
 .TP
 .B Player
 This controls where the player starts from. There must be exactly one
--- a/xmris.man
+++ b/xmris.man
@@ -1,7 +1,7 @@
-. Copyright (C) 1995, 1994, 1993, 1992 Nathan Sidwell
-. RCS $Id: xmris.man 1.3 Tue, 16 Mar 1999 11:28:16 +0000 nathan $
+.\" Copyright (C) 1995, 1994, 1993, 1992 Nathan Sidwell
+.\" RCS $Id: xmris.man 1.3 Tue, 16 Mar 1999 11:28:16 +0000 nathan $
 .TH XMRIS 6 "12 December 1995" "X Version 11"
-.IX xmris#(n) "" "\fLxmris\fP(n)"
+.\" .IX xmris#(n) "" "\fLxmris\fP(n)"
 .SH NAME
 xmris - video game for X
 .SH SYNOPSIS
@@ -10,7 +10,7 @@
 .LP
 .B xmsit
 [-option ...] [-toolkitoption ...]
-.IX xmris#(n) "" "\fLxmris\fP(n) \(em video game"
+.\" .IX xmris#(n) "" "\fLxmris\fP(n) \(em video game"
 .SH DESCRIPTION
 .PP
 Mr Is is a version of the Mr Do video arcade game for the X Window
@@ -986,7 +986,7 @@
 .PP
 The visual class name conversion is performed by a standard toolkit
 routine. It accepts only American spelling, the English spelling of
-'grey' and 'colour' are not allowed.
+\fIgrey\fR and \fIcolour\fR are not allowed.
 .SH COPYRIGHT
 Copyright (C) 1995, 1994, 1993, 1992 Nathan Sidwell.
 .SH AUTHOR
--- a/timer.c
+++ b/timer.c
@@ -85,7 +85,7 @@
       sigprocmask(SIG_BLOCK, &temp, (maskp)); \
     }
 #  define signal_release(maskp) sigprocmask(SIG_SETMASK, maskp, (MASK *)NULL)
-#  define signal_pause(maskp) sigpause(*(maskp))
+#  define signal_pause(maskp) sigsuspend(maskp)
 #else
 #  ifdef __hpux /* hpux is a weird mixture of BSD & SYSV */
 /* don't know if this is right */
@@ -286,8 +286,10 @@
 #else
 	{
 #endif
+#if 0
 	  setitimer(ITIMER_REAL, &timer.interval,
 	     (struct itimerval *)NULL);
+#endif
 	}
     }
   return;
@@ -421,6 +423,8 @@
       timer.delay = (tick_t)(tick / TICKTIME);
 #endif /* BUSYWAIT */
       timer.interval.it_value.tv_usec = tick;
+      timer.interval.it_interval.tv_usec = tick;
+    setitimer(ITIMER_REAL, &timer.interval, (struct itimerval *)NULL);
 #endif /* TRASNPUTER */
     }
   value = timer.state;
